PostProcessing/Source/Common/Utils.py

```diff
@@ -0,0 +1,378 @@
+
+    Raises:
+        RuntimeError: please specify a parameter file
+
+    Returns:
+        configparser.ConfigParser: instance of a ConfigParser class that holds the users parameters
+    """
+
+    if len(arg_list) < 2:
+        raise RuntimeError("Please specify a parameter file")
+    
+    config = configparser.ConfigParser()
+    config.read(arg_list[1])
+
+    if __visit_imported: #If we're running VisIt, then these are the parameter sections we expect
+        mandatory_visit_headers = ["Header", "EngineConfig", "VariableData", 
+                            "AnnotationConfig", "PlotConfig", 
+                            "ViewConfig", "Output"]
+    elif __extra_python: #If we're running via python, then these are the parameter sections expect
+        mandatory_visit_headers = ["Header", "VariableData", "PlotConfig", "Output"]
+
+
+    loaded_config_sections = config.sections()  
+
+    for header in mandatory_visit_headers:
+        if header not in loaded_config_sections:
+            raise configparser.NoSectionError("Parameter file error. Missing section: " + header)
+
+    return config
+
+def create_output_dirs(config):
+    """create output directories if they don't exist
+
+    Args:
+        config (configparser.ConfigParser): instance of a ConfigParser class that holds the users parameters
+    """
+    plot_path = config["Output"].get("output_plot_path", fallback = "./plots")
+    movie_path = config["Output"].get("output_movie_path", fallback = "./movies")
+
+    if not os.path.exists(plot_path):
+        os.mkdir(plot_path)
+    if not os.path.exists(movie_path):
+        os.mkdir(movie_path)
+
+def plot_func_selector(type):
+    """selector function for type of plot to generate
+
+    Args:
+        type (str): string defining type of plot function to choose. Options: '2d' and '3d'
+
+    Raises:
+        ValueError: Invalid plot type
+
+    Returns:
+        func: appropriate plotting function
+    """
+
+    if type == '2d':
+        return Source.TwoD.generate_2dslice_plot
+    elif type == '3d':
+        return Source.ThreeD.generate_3dvolume_plot
+    else:
+        raise ValueError("Invalid plot type. Must be either '2d' or '3d'")
+
+def PlotFiles(config):
+    """	Find all the plot files and return them as list of absolute path strings
+
+    Args:
+        config (configparser.ConfigParser): instance of a ConfigParser class that holds the users parameters
+
+    Raises:
+        FileNotFoundError: Error raised if no plot files were found
+
+    Returns:
+        list: list of plot files
+    """
+
+    #path to hdf5 files plus plot file header string
+    filename_prefix = os.path.join(config["Header"]["hdf5_path"], config["Header"]["plot_header"])
+
+    #get list of file using regex
+    files = glob.glob(filename_prefix+ "*.3d.hdf5")
+
+    #filter the file list to ensure regex-captured files are actual plot files
+    plot_files = [x for x in files if config["Header"]["plot_header"] in x]
```

Comment:
You could also use the `filter` function to express your intent more clearly

Reply:
Ah, thats probably better. added in latest commit